### PR TITLE
Pro 2960 add dependencies to health endpoint

### PR DESCRIPTION
**Pull request by davejones74**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-2960

### Change description ###
Updated to include two new health indicators.
- ccdService
- persistenceService

Please note because the application.yml "management: security: enabled" property is false all the health indicators are exposed in the endpoint. This means the endpoint currently reports on the following in addition to the custom health indicators above.
- mail  
- diskSpace  
- refreshScope
- hystrix
- uk.gov.hmcts.reform.authorisation.healthcheck.ServiceAuth (s2s)

Finally the ccdService health indicator is mapped to the application.yml "services: coreCaseData: enabled:" property which uses the environment variable ${USE_CCD:false}. If this is false the health indicator bean for ccdService will not get created and therefore the check will not take place.

Example JSON being returned: 
{"status":"UP","persistenceService":{"status":"UP"},"mail":{"status":"UP","location":"localhost:2525"},"diskSpace":{"status":"UP","total":499963170816,"free":422914011136,"threshold":10485760},"refreshScope":{"status":"UP"},"hystrix":{"status":"UP"},"uk.gov.hmcts.reform.authorisation.healthcheck.ServiceAuth":{"status":"UP"}}

If an error occurs you get:
{"status":"DOWN","persistenceService":{"status":"DOWN","error":"Connection failed with ResourceAccessException"},"mail":{"status":"UP","location":"localhost:2525"},"diskSpace":{"status":"UP","total":499963170816,"free":422912684032,"threshold":10485760},"refreshScope":{"status":"UP"},"hystrix":{"status":"UP"},"uk.gov.hmcts.reform.authorisation.healthcheck.ServiceAuth":{"status":"UP"}}

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/services/submit/health/SubmitHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            log.error(hsce.getMessage(), hsce);
+            return getHealthWithDownStatus(url, hsce.getMessage(),
+                    "HttpStatusCodeException - HTTP Status: " + hsce.getStatusCode().value());
+        } catch (UnknownHttpStatusCodeException uhsce) {
+            log.error(uhsce.getMessage(), uhsce);
+            return getHealthWithDownStatus(url, uhsce.getMessage(), "UnknownHttpStatusCodeException - " + uhsce.getStatusText());
+        }
+
+        if (responseEntity != null && !responseEntity.getStatusCode().equals(HttpStatus.OK)) {
+        	System.out.println("stage BAD");
```
Comment (nigeldunne) [inline src/main/java/uk/gov/hmcts/probate/services/submit/health/SubmitHealthIndicator.java]: Can we change this to a log statement?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -37,24 +45,27 @@
...
+      formdata.url:  ${services.persistence.baseUrl}/formdata
       #service.url:  ${probate_persistence_service_url}:${probate_persistence_service_port}/submissions  #Till byron fixed side way communcation of app
-      submissions.url:  http://localhost:8282/submissions
-      sequenceNumber.url:  http://localhost:8282/sequence-number
+      submissions.url:  ${services.persistence.baseUrl}/submissions
+      sequenceNumber.url:  ${services.persistence.baseUrl}/sequence-number
 
     coreCaseData:
-      url: https://case-data-app.dev.ccd.reform.hmcts.net:4451/citizens/%s/jurisdictions/PROBATE/case-types/GrantOfRepresentation
+      baseUrl: https://case-data-app.dev.ccd.reform.hmcts.net:4451
```
Comment (nigeldunne) [inline src/main/resources/application.yml]: Check references to the property

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -37,24 +45,27 @@
...
+      url: ${services.coreCaseData.baseUrl}/citizens/%s/jurisdictions/PROBATE/case-types/GrantOfRepresentation
       enabled: ${USE_CCD:false}
 
   auth.provider.service.client:
-    baseUrl: http://localhost:4502
+    #baseUrl: http://localhost:4502
+    baseUrl: http://betadevaccidams2slb.reform.hmcts.net
```
Comment (nigeldunne) [inline src/main/resources/application.yml]: I don't think we should change this. 
It's overridden when it's deployed, and probably used when testing locally with the stub.

File: src/main/java/uk/gov/hmcts/probate/services/submit/health/SubmitHealthIndicator.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.client.UnknownHttpStatusCodeException;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
```
Comment (nigeldunne) [inline src/main/java/uk/gov/hmcts/probate/services/submit/health/SubmitHealthIndicator.java]: If we're going to introduce lombok maybe we should do it consistently across the whole project

File: src/main/resources/application.yml (hunk 2 above)
Comment (davejones74) [reply src/main/resources/application.yml]: This change should be ok. The property services.coreCaseData.url still exists in the application.yml file. For each CNP environment the services.coreCaseData.baseUrl would need to change so the health endpoint and ccd service can be reached. Praveen is aware of this.

**Pull request merged**